pkg/sass-parser/lib/src/expression/number.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import * as postcss from 'postcss';
+
+import {LazySource} from '../lazy-source';
+import type * as sassInternal from '../sass-internal';
+import * as utils from '../utils';
+import {Expression} from '.';
+
+/**
+ * The initializer properties for {@link NumberExpression}.
+ *
+ * @category Expression
+ */
+export interface NumberExpressionProps {
+  value: number;
+  unit?: string;
```

Comment:
FYI - I was thinking of adding numerator/denominator units (this is what prompted the other PR https://github.com/sass/dart-sass/pull/2375)

However, a user can never write `1px*px`, or in other words, Sass/SCSS code can never have a complex unit. It is used by Sass internally, yes, but it's not valid Sass code. We can only ever have a single unit here, or no unit at all. Is this correct?

Reply:
Yep, that's right. No need for complex unit literals.